backend/server.js

```diff
@@ -0,0 +1,224 @@
+          if (err) throw err;
+          console.log(result);
+          console.log("account created");
+          res.status(200);
+          res.send(
+            JSON.stringify({ success: true, result: "account created" })
+          );
+          return;
+        }
+      );
+    }
+  );
+});
+
+//---------------------------------------------------------------------------
+//AutoLogin
+app.post("/auto-login", upload.none(), (req, res) => {
+  let loggedIn = false;
+  console.log(req.cookies.sid);
+  Collection_Sessions.find({ sid: req.cookies.sid }).toArray(
+    (err, foundSessionArr) => {
+      if (foundSessionArr.length < 1) {
+        return;
+      }
+      let foundUser = foundSessionArr[0];
+      console.log("session match");
+      console.log("logged in user " + foundUser.username);
+      loggedIn = true;
+      res.status(200);
+      res.send(
+        JSON.stringify({
+          success: true,
+          result: "logged in as " + foundUser.username
+        })
+      );
+    }
+  );
+});
+//LOGIN
+app.post("/login", upload.none(), (req, res) => {
+  if (req.body === undefined) {
+    console.log("no data with login request");
+    res.status(400);
+    res.send(
+      JSON.stringify({ success: false, result: "login data not recieved" })
+    );
+    return;
+  }
+  console.log("login request recieved || user: " + req.body.username);
+  Collection_LoginInfo.find({ username: req.body.username }).toArray(
+    (err, foundUserArr) => {
+      if (foundUserArr.length < 1) {
+        res.send(
+          JSON.stringify({
+            success: false,
+            result: "username or password was incorrect"
+          })
+        );
+      }
+      const foundUser = foundUserArr[0];
+      if (passwordHash.verify(req.body.password, foundUser.hashedPassword)) {
+        console.log("password match");
+        const sessionId = generateRandomString(20);
+
+        Collection_Sessions.find({ username: req.body.username }).toArray(
+          (err, SessionArr) => {
+            if (SessionArr.length < 1) {
+              Collection_Sessions.insertOne({
+                sid: sessionId,
+                username: req.body.username
+              });
+              return;
+            }
+            let query = { username: req.body.username };
+            let updatedSession = {
+              $set: {
+                sid: sessionId,
+                username: req.body.username
+              }
+            };
+            Collection_Sessions.updateOne(
+              query,
+              updatedSession,
+              (err, result) => {
+                if (err) throw err;
+                console.log("session updated");
+              }
+            );
+          }
+        );
+
+        res.cookie("sid", sessionId);
+        res.status(200);
+        res.send(
+          JSON.stringify({
+            success: true,
+            reesult: "logged in as " + req.body.username
+          })
+        );
+        return;
```

Comment:
This technically works - you don't _need_ to wait for the database to store the session before saving the session ID in the cookie and returning - but it's generally better practice to complete all database work before returning to the client*. This way, you never have issues where the user gets a 200 OK but actually something went wrong, and now the client and server are out of sync

* There are times when the amount of database work is simply too long to wait. For these situations, you typically use a queueing system like RabbitMQ. Don't worry too much about this now, just mentioning it so you know there _are_ solutions for this kind of thing